internal/resource/http/http_test.go

```diff
@@ -391,3 +387,100 @@ func TestResourceDeleteHandler(t *testing.T) {
 		require.ErrorContains(t, err, "resource not found")
 	})
 }
+
+func TestResourceListHandler(t *testing.T) {
+	aclResolver := &resourceSvc.MockACLResolver{}
+	aclResolver.On("ResolveTokenAndDefaultMeta", testACLTokenArtistListPolicy, mock.Anything, mock.Anything).
+		Return(svctest.AuthorizerFrom(t, demo.ArtistV2ListPolicy), nil)
+	aclResolver.On("ResolveTokenAndDefaultMeta", testACLTokenArtistWritePolicy, mock.Anything, mock.Anything).
+		Return(svctest.AuthorizerFrom(t, demo.ArtistV2WritePolicy), nil)
+
+	client := svctest.RunResourceServiceWithACL(t, aclResolver, demo.RegisterTypes)
+
+	r := resource.NewRegistry()
+	demo.RegisterTypes(r)
+
+	handler := NewHandler(client, r, parseToken, hclog.NewNullLogger())
+
+	t.Run("should return MethodNotAllowed", func(t *testing.T) {
+		rsp := httptest.NewRecorder()
+		req := httptest.NewRequest("PUT", "/demo/v2/artist?partition=default&peer_name=local&namespace=default", strings.NewReader(""))
+
+		req.Header.Add("x-consul-token", testACLTokenArtistListPolicy)
+
+		handler.ServeHTTP(rsp, req)
+
+		require.Equal(t, http.StatusMethodNotAllowed, rsp.Result().StatusCode)
+	})
+
+	t.Run("should be blocked if the token is not authorized", func(t *testing.T) {
+		rsp := httptest.NewRecorder()
+		req := httptest.NewRequest("GET", "/demo/v2/artist?partition=default&peer_name=local&namespace=default", strings.NewReader(""))
+
+		req.Header.Add("x-consul-token", testACLTokenArtistWritePolicy)
+
+		handler.ServeHTTP(rsp, req)
+
+		require.Equal(t, http.StatusForbidden, rsp.Result().StatusCode)
+	})
+
+	t.Run("should return list of resources", func(t *testing.T) {
+		resource := createResource(t, handler)
+
+		rsp := httptest.NewRecorder()
+		req := httptest.NewRequest("GET", "/demo/v2/artist?partition=default&peer_name=local&namespace=default", strings.NewReader(""))
+
+		req.Header.Add("x-consul-token", testACLTokenArtistListPolicy)
+
+		handler.ServeHTTP(rsp, req)
+
+		require.Equal(t, http.StatusOK, rsp.Result().StatusCode)
+
+		var result map[string]any
+		require.NoError(t, json.NewDecoder(rsp.Body).Decode(&result))
+
+		resources, _ := result["resources"].([]any)
+		require.Len(t, resources, 1)
+
+		require.Equal(t, resource, resources[0])
+
+		// clean up
+		deleteResource(t, handler)
+	})
+
+	t.Run("should return empty list when no resources are found", func(t *testing.T) {
+		rsp := httptest.NewRecorder()
+		req := httptest.NewRequest("GET", "/demo/v2/artist?partition=default&peer_name=local&namespace=default", strings.NewReader(""))
+
+		req.Header.Add("x-consul-token", testACLTokenArtistListPolicy)
+
+		handler.ServeHTTP(rsp, req)
+
+		require.Equal(t, http.StatusOK, rsp.Result().StatusCode)
+
+		var result map[string]any
+		require.NoError(t, json.NewDecoder(rsp.Body).Decode(&result))
+
+		resources, _ := result["resources"].([]any)
+		require.Len(t, resources, 0)
+	})
+
+	t.Run("should return empty list when name prefix matches don't match", func(t *testing.T) {
```

Comment:
do you think we need to add one more test to confirm the prefix name functionality works?

Reply:
Ya sure, added!